.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,34 @@
+name: Tag repo; build and publish assets
+on:
+  create:
+    tags: ['v*']
+
+env:
+  IMAGE_NAME: cohortextractor
+
+jobs:
+  build-and-publish-docker-image:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    env:
+      image: ghcr.io/opensafely-core/cohortextractor-v2:latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Get current git tag
+        id: tags
+        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
+
+      - name: Build image
+        run: docker build . --file Dockerfile --tag ${{ env.image }}
+
+      - name: Log into GitHub Container Registry
+        run: docker login https://ghcr.io -u ${{ github.actor }} --password ${{ secrets.DOCKER_RW_TOKEN }}
+
+      - name: Push image to GitHub Container Registry
+        env:
+          VERSION: ${{ steps.tags.outputs.tag }}
+        run: |
+          docker tag $IMAGE_NAME $IMAGE_ID:latest
```

Comment:
This first line is not needed anymore with your explicit `image` variable.

FTR, the reason we used `IMAGE_NAME` is that we don't explicitly tag the build in other scenarios, as we have a multi-stage build Docker file, which tags each separate built image with specific names. Hence this retagging the prod version to publish in these two lines.

Currently, the Dockerfile for cohortextractor-v2 is not multi-stage, but it should be in the future. And therefore we may need to rethink the explicit tagging with `--tag ${{ env.image }}` above. But this is fine for now.

Reply:
So should I remove the first `docker tag …` here? or keep both but with `evn.image`?

Reply:
Oh, do you mean that `--tag ${{ env.image }}` in the `Build image` step is equivalent to the first line here?

Reply:
Yes, there's no need to tag with the prod tag as you've already done it via the expliciti --tag to `docker build`

Reply:
Gotcha, done.